app/scripts/main.js

```diff
@@ -0,0 +1,127 @@
+(function() {
+  'use strict';
+
+  var allCircles = []
+    , canvas = document.getElementById("header-canvas")
+    , context = canvas.getContext("2d")
+    , opacity = 1
+    , colors = [
+        'rgba(255, 255, 255,' + opacity + ')',
+        'rgba(200, 200, 200,' + opacity + ')',
+        'rgba(100, 100, 100,' + opacity + ')'
+      ]
+    , minCircleSize = 0.5
+    , maxCircleSize = 2
+    , numCircles = 100
+    , minSpeed = -0.5
+    , maxSpeed = 0.5
+    , speedFactor = 10
+    , dpi = window.devicePixelRatio || 1;
+
+  function initializeCircleArray() {
+    for (var i = 0; i < numCircles ; i++) {
+      var color = Math.floor(Math.random() * (colors.length))
+        , left = Math.floor(Math.random() * (canvas.width +1))
+        , top = Math.floor(Math.random() * (canvas.height + 1))
+        , size = Math.floor(Math.random() * (maxCircleSize - minCircleSize + 1)) + minCircleSize
+        , leftSpeed = (Math.floor(Math.random() * (maxSpeed - minSpeed + 1)) + minSpeed) / speedFactor
+        , topSpeed = (Math.floor(Math.random() * (maxSpeed - minSpeed + 1)) + minSpeed) / speedFactor;
+
+      var circle = {
+        color: color,
+        left: left,
+        top: top,
+        size: size,
+        leftSpeed: leftSpeed,
+        topSpeed: topSpeed,
+      };
+
+      allCircles.push(circle);
```

Comment:
Should just push the object literal directly instead of putting it in a variable and then pushing the variable